micro-services/broker-selection/README.md

```diff
@@ -0,0 +1,31 @@
+# Broker Selection
+
+## Terms
+
+Before moving on, you should understand these two terms of communication:
+
+*Request/Reply* communication is when you expected the destination service
+reply immediately after processed. The simplest example is HTTP request.
+
+![Request Reply Communication](https://www.plantuml.com/plantuml/svg/POxD2i9038JlVOgyGF4k8fLwyI12Ym-mR8Uok9lM91Nwzcwf-8EN367cOwPISs9JYjJWFa4jwnXdgeUK9WxUMHYuq5lWqObBrDeSGMeI_49zEmlF7cUekXw767fEO2EV3wIuAgI5F887bNkwIFVvj-_sMk0Cg_k3xCf_90_TDD1nOCMBxDOKdOv51Uxon2S0 "Request Reply Communication")
+
+*Distributing* communication is when you don't need reply from the
+destination service. Most of the time distributing communication is async.
+
+![Request Reply Communication](https://www.plantuml.com/plantuml/svg/ZP3D2i9038JlVOezwa6yYuZINZo8OF41rkrG5tVJcgH2tzuj1VzwyP8GPhvao5RKidOnc1Fk7KHJRqDlyYR6ZhopwXlAvEQfjMoZARKFCABitK1zAWDF3WU8s1gxu6kA83Kk5N1s5d09VdW7izYO8rVlUVMwdZEiObIpBqzYIbXsrpSqcglHRMQ57t9TEiLyWkx8Xbwz-ml-ZBfUJAEc1orFaBOXmS-UVP5jGLLwpHC0 "Request Reply Communication")
+
+## Decision Tree
+
+![Broker Selection](https://www.plantuml.com/plantuml/svg/bPB1ReCm343FvIjyRBdilXtMhZtjK6rbKhLZoWn49J6BdLNy_GoCegwgfba28YVoUpPPifWWqTNm5EY80JAiCHTBFkbt4bl0V4FGKeIJyG92mEW_eB0rWbI1Obc1GsPJ8Xpw6eyBtK0FytVaHG8QzscwheWOeTozhDFqeS-oK9DWhTdvbeQ37MrV6O6ZLGOLa9DpqTlST4fmG3cXLbQRe1hXo-R8PzmEFoEotEsmgTj5CflD9kvszPPrN4tss0use3UM9TX35ElBGM8GqQwHVqWj2peWNxSGejS5D06TPRnkTOnBD3oZP1BGkB7vCSusaz4uZCdbwXSuOxteqz5_JN-Sossr_-XTATxMl2oxscUGlMP_rjExP8jaMWtlfNxqKdu3 "Broker Selection")
```

Comment:
I think we should save image to our server because we don't sure if 'www.plantuml.com' will be shutdown or change url format.

Reply:
You can use proxy provided by them and use our saved diagram in git like this krubb:

```
https://www.plantuml.com/plantuml/proxy?src=https://raw.githubusercontent.com/C0D1UM/technical-standard/main/devops/git-flow/flow-selection.plantuml
```

This came from Git Flow topic krubb.

Reply:
I did some search and people recommend using proxy as Earth suggested
https://stackoverflow.com/questions/32203610/how-to-integrate-uml-diagrams-into-gitlab-or-github
https://blog.anoff.io/2018-07-31-diagrams-with-plantuml/

Reply:
So I'll change to using proxy